docs/howtos/when-to-use-what-in-the-ffi.md

```diff
@@ -0,0 +1,133 @@
+
+# When to use what method of passing data between Rust and Java/Swift
+
+There are a bunch of options here. For the purposes of our discussion,
+there are two kinds of values you may want to pass over the FFI.
+
+1. Types with identity
+2. Plain ol' data.
+
+## Types with identity
+
+Examples of this are things like database connections, the FirefoxAccounts
+struct, etc. These types are complex, implemented in rust, and it's not
+unreasonable for them to come to Java/Kotlin as a type representing a
+resource (e.g. implementing `Closable`/`AutoClosable`).
```

Comment:
To expand on my above question, this is a good explanation, I'm curious if 'Types with Identity' is an established term somewhere, or if it was chosen for a different reason. I have seen complex types of this type identified in various ways, "Stateful Types", etc. Mainly curious if there's a more widely recognized term for it here.

Reply:
Yeah, it's a term I think I picked up from when I did more functional programming. I know Clojure in particular makes a distinction between value and identity, which is probably why I called them this.

I'm using it to mean something more broad than just resource types (and the terminology is somewhat muddled, as for our case, they necessarily must be resource types when ported to Kotlin, due to the deprecation of `finalize`).

Stateful types is a lot closer to what I mean, but I'll have a think about what a better term for this is.